### PR TITLE
Let secret-operator handle PKCS#12 conversion

### LINKED ISSUE #386: Let secret-operator handle PKCS#12 conversion
We do not need to create key and truststores ourselves and instead let the secret operator provision that.

Example PR in zookeeper for reference https://github.com/stackabletech/zookeeper-operator/pull/695

currently blocked by (we need a password in the keystore):
https://github.com/stackabletech/operator-rs/pull/645
https://github.com/stackabletech/secret-operator/pull/314

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- bumped operator-rs
- use secret operator pkcs12 store with password

fixes  https://github.com/stackabletech/hdfs-operator/issues/386

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by maltesander** [comment]
test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/97/

File: rust/operator/src/container.rs [hunk 1]
```diff
@@ -475,11 +426,7 @@ impl ContainerConfig {
         args.push_str(&self.create_config_directory_cmd());
         args.push_str(&self.copy_config_xml_cmd());
 
-        // We can't influence the order of the init containers.
-        // Some init containers - such as format-namenodes - need the tls certs or kerberos tickets, so let's wait for them to be properly set up
-        if hdfs.has_https_enabled() {
-            args.push_str(&Self::wait_for_trust_and_keystore_command());
-        }
+        // Some init containers - such as format-namenodes - need the kerberos tickets, so let's wait for them to be properly set up
```
Comment (sbernauer) [inline rust/operator/src/container.rs]:
```suggestion
        // This env var is required for reading the core-site.xml
```

File: rust/operator/src/container.rs [hunk 2]
```diff
@@ -56,6 +53,10 @@ use stackable_operator::{
 use std::{collections::BTreeMap, str::FromStr};
 use strum::{Display, EnumDiscriminants, IntoStaticStr};
 
+pub(crate) const KEYSTORE_DIR: &str = "/stackable/keystore";
+pub(crate) const KEYSTORE_VOLUME_NAME: &str = "keystore";
+pub(crate) const KEYSTORE_PASSWORD: &str = "changeit";
```
Comment (sbernauer) [inline rust/operator/src/container.rs]: NIT: `KEY_AND_TRUSTSTORE_PASSWORD` would be more expressive

**Review by sbernauer — CHANGES_REQUESTED** [review]
YEAH!

File: rust/operator/src/container.rs (hunk 2 above)
Comment (maltesander) [reply rust/operator/src/container.rs]: https://github.com/stackabletech/hdfs-operator/pull/392/commits/32dc916bc9b95727493d1c52f8d270dd962adf14

**Review by sbernauer — APPROVED** [review]
I think i would prefer `TLS_STORES_DIR` over `TLS_STORE_DIR`, sorry!
Other than that LGTM

**Pull request merged**